src/placeos-models/edge.cr

```diff
@@ -29,10 +33,12 @@ module PlaceOS::Model
       foreign_key: "edge_id",
     )
 
-    before_save :encrypt!
+    def valid?(test : String) : Bool
+      Encryption.decrypt(string: secret, id: salt, level: ENCRYPTION_LEVEL) == test
```

Comment:
May be safer to encrypt the input string and compare, rather than popping a decrypted value into mem.

Reply:
i thought so too, but the iv is random. i'll look into how we've implemented encryption